deploy/testenv/docker-compose.yaml

```diff
@@ -131,6 +138,72 @@ services:
     deploy:
       replicas: 1
 
+
+  #  Real-time Dashboards
+  #  ------------------------------
+  grafana:
+    depends_on:
+      - loki
+    image: grafana/grafana:10.2.2
+    ports:
+      - "3000:3000"
+    environment:
+      GF_AUTH_ANONYMOUS_ENABLED: true
+      GF_AUTH_ANONYMOUS_ORG_ROLE: Admin
+      GF_PATHS_PROVISIONING: '/app.cfg/provisioning'
+    volumes:
+        - ./grafana:/app.cfg
+
+
+  #  Logging with Loki
+  #  ------------------------------
+  loki:
+      image: grafana/loki:2.9.0
+      ports:
+        - "3100:3100"
+      volumes:
+        - ./loki:/etc/loki
+      command: -config.file=/etc/loki/loki-local-config.yaml
+
+  promtail:
```

Comment:
What does this do?

Reply:
Mounts the configuration file of loki. For greater control, the configuration is managed externally.